netcore/System.Private.CoreLib/src/System.Reflection.Emit/DynamicMethod.cs

```diff
@@ -316,14 +317,25 @@ internal override int GetParametersCount ()
 		}
 
 		public override string ToString () {
-			string parms = String.Empty;
-			ParameterInfo[] p = GetParametersInternal ();
-			for (int i = 0; i < p.Length; ++i) {
+			var paramsSb = new StringBuilder ();
+			ParameterInfo [] parameters = GetParametersInternal ();
+			for (int i = 0; i < parameters.Length; ++i) {
+				ParameterInfo p = parameters [i];
 				if (i > 0)
-					parms = parms + ", ";
-				parms = parms + p [i].ParameterType.Name;
+					paramsSb.Append (", ");
+
+				paramsSb.Append (p.ParameterType.IsPrimitive ? p.ParameterType.Name : p.ParameterType.ToString ());
```

Comment:
Same nit about `TypedReference` as below. ByRef handling is also missing (https://github.com/dotnet/coreclr/blob/9773db1e7b1acb3ec75c9cc0e36bd62dcbacd6d5/src/System.Private.CoreLib/src/System/Reflection/MethodBase.CoreCLR.cs#L73-L77) but the usability of that code is questionable.